crates/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -176,8 +176,21 @@ struct Context<'f> {
     branch_ends: HashMap<BasicBlockId, BasicBlockId>,
 
     /// Maps an address to the old and new value of the element at that address
+    /// These only hold stores for one block at a time and is cleared
+    /// between inlining of branches.
     store_values: HashMap<ValueId, Store>,
 
+    /// Maps an address to the old and new value of the element at that address
+    /// The difference between this map and store_values is that this stores
+    /// the old and new value of an element from the outer block whose jmpif
+    /// terminator is being flattened.
+    ///
+    /// This map persists throughout the flattening process, where addresses
+    /// are overwritten as new stores are found. This overwriting is the desired behavior,
+    /// as we want the most update to date value to be stored at a given address as
+    /// we walk through blocks to flatten.
+    outer_block_stores: HashMap<ValueId, Store>,
```

Comment:
It seems like this was added because only storing the most recent store in `store_values` until it is merged caused problems. Is this assumption right? Would this imply that there may be problems if the block `outer_block_stores` refers to is also a nested branch of some other jmpif?

Reply:
I will test further to try and break the assumption, you may be right that I have to do something similar to what is done in `inline_branch` where the `store_values` are cleared before being inserted into again.